### PR TITLE
[e2e tests] Add tests for product bulks edit

### PR DESCRIPTION
**Pull request by adimoldovan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Add a new test for product bulk edit. 
Updated the existing product edit test with `getBy*` locators, test steps and extra checks.

Closes https://github.com/woocommerce/woocommerce-quality/issues/613 .

### How to test the changes in this Pull Request:

`pnpm env:restart && pnpm test:e2e-pw ./tests/e2e-pw/tests/merchant/product-edit.spec.js`

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/product-edit.spec.js [hunk 1]
```diff
@@ -1,64 +1,187 @@
...
+			await page
+				.getByLabel( 'Regular price ($)' )
+				.fill( newProduct.regularPrice );
+			await page
+				.getByLabel( 'Sale price ($)' )
+				.fill( newProduct.salePrice );
+		} );
+
+		await test.step( 'publish the updated product', async () => {
+			await page.locator( '#publish' ).click();
```
Comment (lanej0) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/product-edit.spec.js]: Could probably use a `page.getByRole( )` here?  Locator isn't bad, we just usually opt for user-facing.

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/product-edit.spec.js [hunk 2]
```diff
@@ -1,64 +1,187 @@
...
+			await page.locator( '#doaction' ).click();
+
+			await expect(
+				await page.locator( '#bulk-titles-list li' ).count()
+			).toEqual( products.length );
+		} );
+
+		await test.step( 'update the regular price', async () => {
+			await page
+				.locator( 'select[name="change_regular_price"]' )
```
Comment (lanej0) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/product-edit.spec.js:127]: Can we use a `page.getByRole()` for some of these as well?

**Review by lanej0 — APPROVED** [review]

**Comment by lanej0** [comment]
Just a couple of notes on locators.  I don't know if there were other options, so approving -- but just pointing out in case there is another option!

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/product-edit.spec.js (hunk 2 above)
Comment (adimoldovan) [reply plugins/woocommerce/tests/e2e-pw/tests/merchant/product-edit.spec.js:127]: Actually no, `getByRole` was not an option in this case.

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/product-edit.spec.js (hunk 1 above)
Comment (adimoldovan) [reply plugins/woocommerce/tests/e2e-pw/tests/merchant/product-edit.spec.js]: Yeah, I missed that one, good catch. Updated with https://github.com/woocommerce/woocommerce/pull/44039/commits/6d0bdf07a19dc2b9865d1cbea78943a9cab5bd48

**Pull request merged**